modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -44,76 +44,145 @@ class Instrument_Manager extends \NDB_Menu_Filter
     {
         $config        = \NDB_Config::singleton();
         $this->path    = $config->getSetting("base");
-        $this->headers = array(
+        $this->headers = [
                           'Instrument',
                           'Instrument_Type',
                           'Table_Installed',
                           'Table_Valid',
                           'Pages_Valid',
-                         );
+                         ];
         $this->query   = " FROM test_names t";
-        $this->columns = array(
+        $this->columns = [
                           't.Test_name as Instrument',
                           "'x' as Instrument_Type",
                           "'x' as Table_Installed",
                           "'x' as Table_Valid",
                           "'x' as Pages_Valid",
-                         );
-        $this->validFilters = array('');
-        $this->formToFilter = array();
+                         ];
+        $this->validFilters = [''];
+        $this->formToFilter = [];
 
-        $writable = true;
-        if (!is_writable($this->path . "project/instruments")
-            || !is_writable($this->path . "project/tables_sql")
-        ) {
-            $writable = false;
-        } else {
-            $writable = true;
+        // Check to see whether the quatUser is configured.
+        // If this user is not configured, the instrument cannot be
+        // automatically installed as the table will not be able to be
+        // created.
+        $quat_user_is_configured = $this->isQuatUserConfigured();
+        if (!$quat_user_is_configured) {
+            $this->tpl_data['can_install'] = false;
+            $this->tpl_data['feedback']    = "Instrument installation is not "
```

Comment:
how about saying that the quat_user is not configured?

Reply:
I don't think that's a useful message for a front-end user.

Reply:
@johnsaigle The instrument manager module requires the superuser privilege so it will not be 'any' frontend user that receive that message. I am not against the proposed message, I just think that it do not give any clue on how to solve the "probleme" when the situation,e.g. not having quat_user configured, is not intentional.

Reply:
Fair enough. I will change the error message.

Reply:
@xlecours Done (see below). Please re-review when possible.